apps/web/src/components/data-table/data-table-toolbar.tsx

```diff
@@ -22,7 +21,19 @@ export function DataTableToolbar<TData>({
   return (
     <div className="flex flex-wrap items-center justify-between gap-3">
       <div className="flex flex-1 items-center gap-2">
-        <DataTableFilterInput table={table} />
+        {table.getColumn("statusCode") && (
+          <DataTableFacetedFilter
+            column={table.getColumn("statusCode")}
+            title="Filter by Status Code"
+            options={Object.keys(codesDict).map((key) => {
+              const typedKey = key as keyof typeof codesDict;
+              return {
+                label: codesDict[typedKey].code,
+                value: codesDict[typedKey].status_code,
+              };
+            })}
```

Comment:
Is it possible to only display the `codesDict` keys that exist in our data. Supposedly, we only have `200`, we shouldn't display any other option.

It should be possible to access the data via the `table` attribute and reduce the status code.

Reply:
so the status data will come from the url right ?

Reply:
The data we display is coming from the `getResponseList()`. One row has all the props from `tbBuildResponseList` including `statusCode`.

It would be nice to do something like:

```ts
const existingStatusCodes = data.reduce((prev, curr) => {
  if (!prev.includes(curr.statusCode)) {
    return [...prev, curr.statusCode]
  }
}, [])
```

And then only display the existing status codes instead of all of them.

Reply:
ohhh ok gotcha 
thanks for explaining !